### DISCUSSION TITLE
How to debug end criteria issues

### DISCUSSION THREAD

**Discussion by BatchDrake** [thliebig/openEMS-Project#39] [Q&A Usage] [answered]
Hi,

I've been using OpenEMS lately to simulate microstrip filters, and in some cases I observe that the end criteria is never met (-45 dB). Not only that, but past certain energy (in my case, usually around -40 dB), the energy starts to increase quasi-monotonically. My bet is there is something odd with my boundary conditions (`("PML_8", "PML_8"), ("MUR", "MUR"), ("PEC", "MUR"))`) but even after analyzing the electric field dumps with Paraview I still fail to detect any strange behavior. In fact it shows precisely what I expected to see (if it worked): how the fields fade away.

My question is, what is therefore the procedure to debug issues like this?

**Comment by thliebig** [comment, marked as answer]
Hi,
the pml boundary condition is inherently not perfectly stable (like everything else). That means that especially near-fields can cause the PML to get unstable. If the PML is too close to a structure or if you run your simulation too long then near-fields is all that remain and can cause this.
This means just stop at -40, in most cases that should be fine.
Usually if energy gets trapped than it it could be some static charge. Defining lossy material may help then or exciting a Gaussian pulse with less DC part might help too.

regards
Thorsten
